src/misc/rwqueue.cpp

```diff
@@ -244,24 +257,30 @@ std::optional<T> RWQueue<T>::Dequeue()
 }
 
 template <typename T>
-bool RWQueue<T>::BulkDequeue(std::vector<T>& into_target, const size_t num_requested)
+size_t RWQueue<T>::BulkDequeue(std::vector<T>& into_target, const size_t num_requested)
 {
-	constexpr size_t min_items = 1;
-	assert(num_requested >= min_items);
```

Comment:
IMO, it's good not to have to special-case zero-size operations in such containers. I don't like it when something blows up when I pass it an empty container, etc.

Reply:
Yes, it handles a zero-request size without issue; but it does need to lock and unlock, so I've left the assert at least to catch no op spinners :wheel:.

We might as well cut off the no-op spinning as close to the source as possible, because (at least on my system), hundreds of locks and unlocks per millisecond can chew up a lot of CPU time :laughing: 

(I'm looking at Tyrian's startup phase where the mixer thread runs at 100% CPU my system for a couple seconds.. then settles down to roughly matching the main thread! That game hits it very hard :laughing: )